### DISCUSSION TITLE
Speech and Language Processing (3rd ed. draft)

### DISCUSSION THREAD

**Discussion by chengjun** [computational-class/css#60] [Show and tell]
[Dan Jurafsky](http://web.stanford.edu/people/jurafsky/) and [James H. Martin](http://www.cs.colorado.edu/~martin/)

## Speech and Language Processing, 3rd

Speech and Language Processing (3rd ed. draft) [Dan Jurafsky](http://web.stanford.edu/people/jurafsky/) and [James H. Martin](http://www.cs.colorado.edu/~martin/)

Here's our Jan 7, 2023 draft!

This draft is mostly a bug-fixing and restructuring release, there are no new chapters. The restructuring moves the applications section earlier, reflecting how we and others tend to teach NLP, and combines the linguistic structure chapters in one section.

We've put up a [list here](https://web.stanford.edu/~jurafsky/slp3/thanks.html) of the amazing people who have sent so many fantastic suggestions and bug-fixes for improving the book. We are really grateful to all of you for your help, the book would not be possible without you!

Individual chapters are below; [here is a single pdf of all the chapters in the Jan 7, 2023 draft of the book so far!](https://web.stanford.edu/~jurafsky/slp3/ed3book_jan72023.pdf)

Feel free to use the draft chapters and slides in your classes, the resulting feedback we get from you makes the book better!

As always, typos and comments very welcome (just email slp3edbugs@gmail.com and let us know the date on the draft)! (Don't bother reporting missing refs due to cross-chapter cross-reference problems in the indvidual chapter pdfs, those are fixed in the full book draft)

When will the whole book be finished? Don't ask.

If you need last year's December/Jan 2021/2 draft chapters, they are [here](https://web.stanford.edu/~jurafsky/slp3/old_dec21/);

Chapter Slides Relation to 2nd ed. Part I: Fundamental Algorithms

- 1:Introduction[newly written]2:[Regular Expressions, Text Normalization, Edit Distance](https://web.stanford.edu/~jurafsky/slp3/2.pdf)2: Text Processing [[pptx](https://web.stanford.edu/~jurafsky/slp3/slides/2_TextProc_Mar_25_2021.pptx)] [[pdf](https://web.stanford.edu/~jurafsky/slp3/slides/2_TextProc_Mar_25_2021.pdf)]
- 2: Edit Distance [[pptx](https://web.stanford.edu/~jurafsky/slp3/slides/2_EditDistance_Jan_08_2020.pptx)] [[pdf](https://web.stanford.edu/~jurafsky/slp3/slides/2_EditDistance_Jan_08_2020.pdf)]
- [Ch. 2 and parts of Ch. 3 in 2nd ed.]3:[N-gram Language Models](https://web.stanford.edu/~jurafsky/slp3/3.pdf)3: N-grams [[pptx](https://web.stanford.edu/~jurafsky/slp3/slides/3_LM_Jan_08_2021.pptx)] [[pdf](https://web.stanford.edu/~jurafsky/slp3/slides/3_LM_Jan_08_2021.pdf)]
- [Ch. 4 in 2nd ed.]4:[Naive Bayes and Sentiment Classification](https://web.stanford.edu/~jurafsky/slp3/4.pdf)4: Naive Bayes + Sentiment [[pptx](https://web.stanford.edu/~jurafsky/slp3/slides/4_NB_Apr_4_2021.pptx)] [[pdf](https://web.stanford.edu/~jurafsky/slp3/slides/4_NB_Apr_4_2021.pdf)]
- [new in this edition]5:[Logistic Regression](https://web.stanford.edu/~jurafsky/slp3/5.pdf)5: LR [[pptx](https://web.stanford.edu/~jurafsky/slp3/slides/5_LR_Apr_7_2021.pptx)] [[pdf](https://web.stanford.edu/~jurafsky/slp3/slides/5_LR_Apr_7_2021.pdf)]
- [new in this edition]6:[Vector Semantics and Embeddings](https://web.stanford.edu/~jurafsky/slp3/6.pdf)6: Vector Semantics [[pptx](https://web.stanford.edu/~jurafsky/slp3/slides/6_Vector_Apr18_2021.pptx)] [[pdf](https://web.stanford.edu/~jurafsky/slp3/slides/6_Vector_Apr18_2021.pdf)]
- [new in this edition]7:[Neural Networks and Neural Language Models](https://web.stanford.edu/~jurafsky/slp3/7.pdf)7: Neural Networks [[pptx](https://web.stanford.edu/~jurafsky/slp3/slides/7_NN_Apr_28_2021.pptx)] [[pdf](https://web.stanford.edu/~jurafsky/slp3/slides/7_NN_Apr_28_2021.pdf)]
- [new in this edition]8:[Sequence Labeling for Parts of Speech and Named Entities](https://web.stanford.edu/~jurafsky/slp3/8.pdf)8: POS/NER Intro only [[pptx](https://web.stanford.edu/~jurafsky/slp3/slides/8_POSNER_intro_May_6_2021.pptx)] [[pdf](https://web.stanford.edu/~jurafsky/slp3/slides/8_POSNER_intro_May_6_2021.pdf)]
- [expanded from Ch. 5 in 2nd ed.]9:[RNNs and LSTMs](https://web.stanford.edu/~jurafsky/slp3/9.pdf)
- [new in this edition]10:[Transformers and Pretrained Language Models](https://web.stanford.edu/~jurafsky/slp3/10.pdf)
- [newly written for this edition]11:[Fine-tuning and Masked Language Models](https://web.stanford.edu/~jurafsky/slp3/11.pdf)
- [new in this edition]12:Prompting and Instruct Tuning Part II: NLP Applications13:[Machine Translation](https://web.stanford.edu/~jurafsky/slp3/13.pdf)
- [newly written for this edition, earlier MT was Ch. 25 in 2nd ed.]14:[Question Answering and Information Retrieval](https://web.stanford.edu/~jurafsky/slp3/14.pdf)
- [mostly newly written ; a few sections expanded from parts of Ch 23 in 2nd ed]15:[Chatbots and Dialogue Systems](https://web.stanford.edu/~jurafsky/slp3/15.pdf)15: Dialog [[pptx](https://web.stanford.edu/~jurafsky/slp3/slides/24_Dialogue_May_6_2021.pptx)] [[pdf](https://web.stanford.edu/~jurafsky/slp3/slides/24_Dialogue_May_6_2021.pdf)]
- [mostly new, parts expanded from Ch 24 in 2nd ed]16:[Automatic Speech Recognition and Text-to-Speech](https://web.stanford.edu/~jurafsky/slp3/16.pdf)
- [mostly newly written, ASR and TTS were Chs 8 and 9 in 2nd ed] Part III: Annotating Linguistic Structure17:[Context-Free Grammars and Constituency Parsing](https://web.stanford.edu/~jurafsky/slp3/17.pdf)
- [expanded from Ch. 12 and 13 in 2nd ed.]18:[Dependency Parsing](https://web.stanford.edu/~jurafsky/slp3/18.pdf)
- [new in this edition]19:[Logical Representations of Sentence Meaning](https://web.stanford.edu/~jurafsky/slp3/19.pdf)[Ch. 17 in 2nd ed.]20:Computational Semantics and Semantic Parsing21:[Relation and Event Extraction](https://web.stanford.edu/~jurafsky/slp3/21.pdf)
- [expanded from Ch. 22 in 2nd ed.]22:[Time and Temporal Reasoning](https://web.stanford.edu/~jurafsky/slp3/22.pdf)
- [expanded from Ch. 22 in 2nd ed.]23:[Word Senses and WordNet](https://web.stanford.edu/~jurafsky/slp3/23.pdf)
- [expanded from Ch. 20 in 2nd ed.]24:[Semantic Role Labeling and Argument Structure](https://web.stanford.edu/~jurafsky/slp3/24.pdf)
- [expanded from parts of Ch. 19, 20 in 2nd ed]25:[Lexicons for Sentiment, Affect, and Connotation](https://web.stanford.edu/~jurafsky/slp3/25.pdf)25: Affect [[pptx](https://web.stanford.edu/~jurafsky/slp3/slides/20_Affect_Jan_18_2021.pptx)] [[pdf](https://web.stanford.edu/~jurafsky/slp3/slides/20_Affect_Jan_18_20201.pdf)]
- [new in this edition]26:[Coreference Resolution](https://web.stanford.edu/~jurafsky/slp3/26.pdf)
- [mostly newly written; some sections expanded from parts of Ch 21 in 2nd ed]27:[Discourse Coherence](https://web.stanford.edu/~jurafsky/slp3/27.pdf)
- [mostly new for this edition]28:[Phonetics](https://web.stanford.edu/~jurafsky/slp3/28.pdf)[Ch 7 in 2nd ed] 

Appendix Chapters (will be just on the web)A:[Hidden Markov Models](https://web.stanford.edu/~jurafsky/slp3/A.pdf)B:[Spelling Correction and the Noisy Channel](https://web.stanford.edu/~jurafsky/slp3/B.pdf)C:[Statistical Constituency Parsing](https://web.stanford.edu/~jurafsky/slp3/C.pdf)D:[Context-Free Grammars](https://web.stanford.edu/~jurafsky/slp3/D.pdf)
